warehouse/query-core/src/main/java/datawave/query/tables/edge/EdgeQueryLogic.java

```diff
@@ -148,6 +161,83 @@ public GenericQueryConfiguration initialize(Connector connection, Query settings
         return cfg;
     }
     
+    public String getJexlQueryString(Query settings) throws datawave.query.language.parser.ParseException {
+        // queryString should be JEXl after all query parsers are applied
+        String queryString;
+        String originalQuery = settings.getQuery();
+        
+        originalQuery = this.expandQueryMacros(originalQuery);
+        
+        if (null == originalQuery) {
+            throw new IllegalArgumentException("Query cannot be null");
+        }
+        
+        // Determine query syntax (i.e. JEXL, LUCENE, etc.)
+        String querySyntax = settings.findParameter(QueryParameters.QUERY_SYNTAX).getParameterValue();
+        
+        // enforce mandatoryQuerySyntax if set
+        if (null != this.mandatoryQuerySyntax) {
+            if (org.apache.commons.lang.StringUtils.isEmpty(querySyntax)) {
+                throw new IllegalStateException("Must specify one of the following syntax options: " + this.mandatoryQuerySyntax);
+            } else {
+                if (!this.mandatoryQuerySyntax.contains(querySyntax)) {
+                    throw new IllegalStateException("Syntax not supported, must be one of the following: " + this.mandatoryQuerySyntax + ", submitted: "
+                                    + querySyntax);
+                }
+            }
+        }
+        
+        QueryParser querySyntaxParser = getParser();
+        
+        if (org.apache.commons.lang.StringUtils.isBlank(querySyntax)) {
+            // Default to the class's query parser when one is not provided
+            // Falling back to Jexl when one is not set on this class
+            if (null == querySyntaxParser) {
+                querySyntax = "JEXL";
+            }
+        }
+        if (querySyntax.equals("JEXL")) {
```

Comment:
it is important to note that if we are NOT using JEXL then a parser is required to be defined in the LogicFactory.xml (can be null)
the behavior before accepted LIMITED_JEXL/LIST without the parser because it didn't actually have any parsing functionallity

Reply:
if LUCENE is passed as a syntax how would you want to deal with that?